serial_x.go

```diff
@@ -0,0 +1,116 @@
+// +build !windows
+
+package serial
+
+import (
+	"time"
+	"io/ioutil"
+	"regexp"
+	"strings"
+)
+
+const (
+	devFolder = "/dev"
+	regexFilter = `^(cu|tty)\.?.*`
```

Comment:
I understand that this works in a practical sense for 95% of serial ports.

Is there any documentation that this 1) does not return non-serial ports and 2) returns all the serial ports in the system?

For windows, I understand that the API might be tricky and it might make sense to have a special function for it.

On linux etc, if it's not 100% I'm not sure what the benefit of including the regex in this package is.  It could be wrong and the user could just as easily do the regex in their own own.  The regex might still be wrong or incomplete, but at least they would not get a false sense of security about it coming from the library (where the user should expect things to be complete and well tested).

Thoughts?